src/main/java/com/ecommerce/digital/adapter/output/repository/mysql/UserManagementMySql.java

```diff
@@ -0,0 +1,60 @@
+package com.ecommerce.digital.adapter.output.repository.mysql;
+
+import com.ecommerce.digital.adapter.output.repository.dto.UserDto;
+import com.ecommerce.digital.application.ports.output.UserManagementOutputPort;
+import com.ecommerce.digital.domain.User;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+
+import java.util.List;
+import java.util.Optional;
+
+@Component
+public class UserManagementMySql implements UserManagementOutputPort {
+    @Autowired
+    UserRepository userRepository;
+
+    @Override
+    public User enroll(User user) {
+        UserDto userDto = new UserDto(user);
+        UserDto savedUser = userRepository.save(userDto);
+        return savedUser.toDomain();
+    }
+
+    @Override
+    public User findByMail(String email) {
+        try {
+            return userRepository.findByEmail(email).toDomain();
```

Comment:
Antes de realizar o .toDomain() precisamos verificar que o objeto retornado não seja null
```suggestion
            
            UserDto userDto =  userRepository.findByEmail(email);
            if (userDto != null) return userDto.toDomain() else return null
```

Reply:
feito